### ISSUE TITLE
AvoidNoArgumentSuperConstructorCall: false positive on constructor that throws exception

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#12503]
````
$ cat TestClass.java
public class TestClass extends Base {
  public TestClass() throws Exception {
    super();
  }
}

class Base {
  public Base() throws Exception {
  }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="AvoidNoArgumentSuperConstructorCall" />
    </module>
</module>

$ java -jar checkstyle-10.5.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:3:5: Unnecessary call to superclass constructor with no arguments. [AvoidNoArgumentSuperConstructorCall]
Audit done.
Checkstyle ends with 1 errors.
````

Eclipse complains with `Default constructor cannot handle exception type Exception thrown by implicit super constructor. Must define an explicit constructor` if I don't add Test constructor.

**Comment by rnveach** [comment]
Removing default constructor results in:
````
$ javac --version
javac 11.0.17

$ javac TestClass.java
TestClass.java:1: error: unreported exception Exception in default constructor
public class TestClass
       ^
1 error
````

**Comment by rnveach** [comment]
I am removing this because call to super can be removed and leave in the constructor.

**Comment by rnveach** [comment]
Proof:
````
$ cat TestClass.java
public class Test extends Base {
public Test() throws Exception {
}
}

class Base {
public Base() {
}
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="AvoidNoArgumentSuperConstructorCall" />
    </module>
</module>

$ java -jar checkstyle-10.5.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````
